### PR TITLE
First draft about Map::Tube for 2024 Advent Calendar.

### PR DESCRIPTION
**Pull request by manwar**
Hi Team,

Please review the first draft.

Many Thanks.

Best Regards,
Mohammad

### CODE REVIEW

File: 2024/incoming/have-fun-with-map-tube.pod [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+L<Map::Tube|https://metacpan.org/dist/Map-Tube> was one of my earliest submissions to C<CPAN> and it's a project that's very close to my hear. Over the years, I've had the joy of writing about it and I've spoken about it at many conferences. I consider it my baby though I share it with many amazing contributors who have helped make it sleeker and more powerful. Each day, I'm thankful for the contributions of those who've made this project better. Many thanks also to everyone who has used used the library to build maps for cities around the world. As of today, there are L<41 maps|https://metacpan.org/pod/Map::Tube#MAP-LEADER-BOARD> created by C<11 contributors>.
+
+=head2 Inspiration behind the map
+
+Our first window to open is about inspiration! The journey began with the idea to create a map for the C<London Tube Map>, the famous underground rail network. The first map created was L<Map::Tube::London|https://metacpan.org/dist/Map-Tube-London>. As work on C<Map::Tube::London> progressed, I realized there was an opportunity to build a framework so others could create map for their own cities. And so, C<Map::Tube> was born. So now you know where the term C<Tube> came from!
```
Comment (JJ) [inline 2024/incoming/have-fun-with-map-tube.pod:13]:
```suggestion
Our first window to open is about inspiration! The journey began with the idea to create a map for the C<London Tube>, the famous underground rail network. The first map created was L<Map::Tube::London>. As work on C<Map::Tube::London> progressed, I realized there was an opportunity to build a framework so others could create map for their own cities. And so, C<Map::Tube> was born. So now you know where the term C<Tube> came from!
```

Minor edits.

File: 2024/incoming/have-fun-with-map-tube.pod [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+=head2 The first day's play: Let's get started
+
+It's time to open the first door and dive in! To get started, you'll need to install the framework and at least one map. For today, let's stick with the C<London Tube Map>.
```
Comment (JJ) [inline 2024/incoming/have-fun-with-map-tube.pod:17]: You've said "window" before. Maybe you want to keep it consistent?

File: 2024/incoming/have-fun-with-map-tube.pod [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+=begin code
+
+    $ sudo cpanm -v Map::Tube
```
Comment (JJ) [inline 2024/incoming/have-fun-with-map-tube.pod:21]: nit: Do we really want to encourage people to use `sudo` to install CPAN stuff? It's a personal bone to pick, but I tell my students that devs should never have to go sudo to install stuff they need, if only to encourage them to *never* use system (perl|python|node|whatever)

File: 2024/incoming/have-fun-with-map-tube.pod [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+=begin code
+
+    Baker Street (Bakerloo, Circle, Hammersmith & City, Jubilee, Metropolitan), Finchley Road (Jubilee, Metropolitan), Wembley Park (Jubilee, Metropolitan)
+
+=end code
+
+Nice and easy, right?
+
+If you're feeling a bit lazy and don't want to write the C<4 lines> perl, no worries, there's a shortcut! The L<Map::Tube::CLI|https://metacpan.org/dist/Map-Tube-CLI> distribution comes with a command line utility called C<map-tube>.
```
Comment (JJ) [inline 2024/incoming/have-fun-with-map-tube.pod:49]:
```suggestion
If you're feeling a bit lazy and don't want to write the C<4 lines> of perl, no worries, there's a shortcut! The L<Map::Tube::CLI|https://metacpan.org/dist/Map-Tube-CLI> distribution comes with a command line utility called C<map-tube>.
```

Minor edit

File: 2024/incoming/have-fun-with-map-tube.pod [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    $ sudo cpanm -v Map::Tube::CLI
+
+    $ map-tube --map London --start 'Baker Street' --end 'Wembley Park'
+    Baker Street (Bakerloo, Circle, Hammersmith & City, Jubilee, Metropolitan), Finchley Road (Jubilee, Metropolitan), Wembley Park (Jubilee, Metropolitan)
+
+=end code
+
+=head2 So what's inside the map? Peek behind the curtain
+
+Curious about how it all works? Open the next window to peek inside! There's a package called L<Map::Tube::Cookbook|https://metacpan.org/pod/Map::Tube::Cookbook>, which is part of the C<Map::Tube> distribution that explains everything that you need to know to create a new map.
```
Comment (JJ) [inline 2024/incoming/have-fun-with-map-tube.pod:64]: Back to window...

**Review by JJ — APPROVED** [review]
Great job! Please consider my nits and suggestions

**Comment by manwar** [comment]
@JJ Thanks for the suggestions. Please review it when you get time.

File: 2024/incoming/have-fun-with-map-tube.pod [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+Nice and easy, right?
+
+If you're feeling a bit lazy and don't want to write the C<4 lines> perl, no worries, there's a shortcut! The L<Map::Tube::CLI> distribution comes with a command line utility called C<map-tube>.
+
+Let's repeat today's fun with just one line using the command line tool: C<map-tube>.
+
+=begin code
+
+    $ sudo cpanm -v Map::Tube::CLI
```
Comment (JJ) [inline 2024/incoming/have-fun-with-map-tube.pod:55]: You've dropped the `sudo` before. Maybe you want to keep it consistent too.

**Review by JJ — APPROVED** [review]
:+1: Thanks!

**Review by oalders — COMMENTED** [review]
I love it! ❤️ 

@manwar would you be willing to include an image or two or three to demonstrate what it looks like in `png` format?

**Comment by oalders** [comment]
@JJ would you be interested in reviewing some of the other articles? If so, feel free to self-assign any unreviewed article to yourself and submit a review. I found your work here to be very helpful.

**Comment by JJ** [comment]
Right now I am taking a few days, until tomorrow evening..After that, I could lend a hand

**Comment by oalders** [comment]
I'm thinking Dec 10 for this one.

**Pull request merged**